packages/react/src/hooks/useTokenById.ts

```diff
@@ -16,12 +17,17 @@ export const useTokenById = (
   const [res, setData] = useState<TokenByIdResults | undefined>(undefined);
   const [errorMsg, setError] = useState<string | null>(null);
 
+
+  const contract = contractAddress || mbjs.keys.contractAddress;
+  const validParams = contract && tokenId; 
+
+
   useEffect(() => {
     let isCancelled = false;
 
-    if (loading) {
+    if (loading && validParams) {
```

Comment:
Even in the context of a hook/effect, an early return is always easier to read:

```
if (!something) {
  setError('something needs to be defined')
  return
}
```

vs wrapping later and then noop

Reply:
@sainthiago 

`useEffect expects the return to be a clean up function, so return null is wrong because 1. return is conditional and 2. return is not a function`

this type of code we being doing on omni-site is wrong

more info:
[What is the expected return of `useEffect` used for?](https://stackoverflow.com/questions/56800694/what-is-the-expected-return-of-useeffect-used-for)

[Cleanup useEffect](https://hackernoon.com/cleanup-functions-in-reacts-useeffect-hook-explained)
[Understanding React’s useEffect cleanup function](https://blog.logrocket.com/understanding-react-useeffect-cleanup-function/)

Reply:
We could define the return function in the outer scope if we cared that much, but I think this is fine given it's a hook